### PR TITLE
CDPS-856: Restructuring the database

### PR DESCRIPTION
**Pull request by brightonsbox**
Restructuring to keep individual field history.  This allows us to decouple our table structure from the table structure of NOMIS a bit more.

field_history

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonperson/jpa/FieldHistory.kt [hunk 1]
```diff
@@ -2,91 +2,89 @@ package uk.gov.justice.digital.hmpps.prisonperson.jpa
...
-  var height: Int? = null,
+  @Enumerated(STRING)
+  @Column(updatable = false, nullable = false)
+  var field: PrisonPersonField,
 
-  @Column(name = "weight_kg")
-  var weight: Int? = null,
+  override var valueInt: Int? = null,
 
   val appliesFrom: ZonedDateTime = ZonedDateTime.now(),
```
Comment (davidmiddletonmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonperson/jpa/FieldHistory.kt:32]: Does this mean `appliesFrom` will always match `createdBy`? Does this support NOMIS previous booking changes?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonperson/jpa/FieldHistory.kt [hunk 2]
```diff
@@ -2,91 +2,89 @@ package uk.gov.justice.digital.hmpps.prisonperson.jpa
...
-  )
+  @Enumerated(STRING)
+  val source: Source? = null,
 
-  override fun compareTo(other: PhysicalAttributesHistory) =
+) : FieldValues,
+  Comparable<FieldHistory> {
+
+  override fun compareTo(other: FieldHistory) =
     compareValuesBy(this, other, { it.appliesTo?.toInstant() ?: Instant.MAX }, { it.createdAt }, { it.hashCode() })
```
Comment (davidmiddletonmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonperson/jpa/FieldHistory.kt:45]: Depending on the answer to the above comment, should this include `appliesFrom` as well?

**Review by davidmiddletonmoj — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonperson/jpa/FieldHistory.kt (hunk 1 above)
Comment (brightonsbox) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonperson/jpa/FieldHistory.kt:32]: This is just a default value (only really there to make it easier to write tests if you don't necessarily care about that value, and don't want to explicitly set it)

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonperson/jpa/FieldHistory.kt (hunk 2 above)
Comment (brightonsbox) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonperson/jpa/FieldHistory.kt:45]: I'm just wanting to sort by `appliesTo` really, and then if they're the same then sort by `createdAt`.  I don't think `appliesFrom` gives us anything in terms of sorting.

**Pull request merged**